tests/scenarios/FileSystemTest/PrivateProfileTest.cpp

```diff
@@ -0,0 +1,132 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Tim Mangan - TMurgent Technologies, LLP. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include <filesystem>
+#include <functional>
+
+#include <test_config.h>
+
+#include "common_paths.h"
+#include <iostream>
+#include <fstream>
+
+
+int PrivateProfileTests()
+{
+	int result = ERROR_SUCCESS;
+    int testResult;
+	auto packageFilePath = g_packageRootPath / g_packageFileName;
+	auto otherFilePathIntl = g_packageRootPath / L"TèƨƭFïℓè.ini";
+    auto otherFilePath = g_packageRootPath / L"TestIniFile.ini";
+
+    std::wstring badValue = L"BadValue";
+    std::wstring updatedValue = L"UpdatedValue";
+    int badValueInt = 666;
+
+    clean_redirection_path();
+
+    int   testLen = 2048*sizeof(wchar_t);
+    wchar_t* buffer = (wchar_t *)malloc(testLen);
```

Comment:
Unless you have a need for this to be a C-style string can you please use an std::wstring?

Reply:
This is exactly what a legacy app would do when calling GetPriviate..., isn't that what we want to test?

Reply:
Addressed in Commit 1bdc01a3

Reply:
I see your point.  If you want you can change it back to a C-Style string.  Or, if you want to go above and beyond, you can test both.